File: linked-list-cycle/linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        
        if (head == null)
            return false;
        
        ListNode tort = head;
        ListNode hare = head.next;
        
        while (tort != null && hare != null) {
            if (tort == hare)
                return true;
            else if (hare.next == null)
                return false;
            tort = tort.next;
            hare = hare.next.next;
        } //while
        
        return false;
    } //hasCycle
} //Solution

File: LeetCode/141-linked-list-cycle/linked-list-cycle.cpp

class Solution {
public:
    bool hasCycle(ListNode *head) {
        if (head == nullptr) return false;
        ListNode* run = head->next;
        ListNode* walk = head;

        while (run != nullptr && run->next != nullptr) {
            if (run == walk) return true;
            run = run->next->next;
            walk = walk->next;
        } //while

        return false;
    } //hasCycle
};